### DISCUSSION TITLE
Hi

### DISCUSSION THREAD

**Discussion by Gameidiaz** [excaliburjs/Excalibur#2767] [General]
Will the Excalibur js API will ever change because I don’t want it to change

**Comment by eonarheim** [comment]
Hi @spedy217!

Great question, Excalibur should stay pretty much the same between now and v1 (which will be sometime in the next 6 months)! Before v1 We'll have some release candidates where we can discuss any API changes/tweaks we want to change

We don't have anything too major planned, it's mostly going be focused improving some complexity and refactoring.

Here's roughly the changes that are planned:
* Constructor updates to option bag style `new Thing({...})` instead of long parameter lists
* Particles will be refactored to be more in line with the current API
* Particles will potentially be refactored to be GPU based (for speed!)
* The low level ECS API will be simplified, but will operate pretty much the same way (folks shouldn't notice anything unless you're building custom systems)
* Static globals will be refactored to be actual members on Engine/Scene, for example  `ex.Physics` 
* Graphics component updates to move layer into a separate graphics type
* Resource loading and scene specific resources
* Engine.start overload to take a start scene

We have more features planned too documented here https://github.com/excaliburjs/Excalibur/issues/1161 

To over emphasize, we really like the current API too! Except for some small corners :)

**Reply by eonarheim** [reply]
Once v1 is out, we'll do our best to follow semver and there will be no deliberate breaking changes 😄 

There are no plans anytime soon for a v2
